lib/jobs/ucs-catalog.js

```diff
@@ -44,7 +44,10 @@ function UcsCatalogJobFactory(
             options,
             context,
             taskId);
-        this.node = this.context.target;
+        this.nodes = (this.context.physicalNodeList || []).concat(this.context.logicalNodeList);
```

Comment:
From your implementation, some nodes are extracted from node relations, so there may be some duplicated nodes in `this.nodes`. You need to remove those duplicated nodes.